server/src/main/java/org/elasticsearch/index/mapper/MappedFieldType.java

```diff
@@ -632,6 +634,25 @@ public void validateMatchedRoutingPath(String routingPath) {
         );
     }
 
+    /**
+     * This method is used to support _field_caps when include_fields_with_no_value is set to
+     * {@code false}. In that case we return only fields with value in an index. This method
+     * gets as input FieldInfos and returns if the MappedFieldType#name is present. This method
+     * needs to be overwritten for special cases where Lucene field name do not match user
+     * defined field name two example are rank_feature that indexes into the _feature Lucene
```

Comment:
where fields don't have footprint in Lucene or their name differs from `name()` ?